File: solutions/Leetcode_303/leetcode_303.py
from typing import List


class NumArray:
    def __init__(self, nums: List[int]):
        self._range_sum = [0]
        for x in nums:
            self._range_sum.append(self._range_sum[-1] + x)

    def sum_range(self, left: int, right: int):
        return self._range_sum[right+1] - self._range_sum[left]

File: solutions/Leetcode_303/leetcode_303.cpp
#include <vector>

class NumArray
{
public:
    NumArray(std::vector<int> &nums)
    {
        for (int x : nums)
            _range_sums.push_back(_range_sums.back() + x);
    }

    int sumRange(int left, int right)
    {
        return _range_sums[right + 1] - _range_sums[left];
    }

private:
    std::vector<int> _range_sums = {0};
};
